### PR TITLE
[CHERRY-PICK] src: common: update itt to 3.23.0

### PR DESCRIPTION
**Pull request by EgorDuplensky**
### Details:
- ITT fix for lib loading on windows
- OpenVINO PR - https://github.com/openvinotoolkit/openvino/pull/13910

### Tickets:
- 93408

### CODE REVIEW

File: src/common/ittnotify/ittnotify_static.c [hunk 1]
```diff
@@ -62,7 +62,7 @@ static const char api_version[] = API_VERSION "\0\n@(#) $Revision$\n";
 #define ITT_ATTRIBUTE_FALLTHROUGH [[gnu::fallthrough]]
 #elif HAS_CPP_ATTR(clang::fallthrough)
 #define ITT_ATTRIBUTE_FALLTHROUGH [[clang::fallthrough]]
-#elif HAS_GNU_ATTR(fallthrough) && !defined(__INTEL_COMPILER)
+#elif HAS_GNU_ATTR(fallthrough) && !__INTEL_COMPILER
 #define ITT_ATTRIBUTE_FALLTHROUGH __attribute__((fallthrough))
```
Comment (usstq) [inline src/common/ittnotify/ittnotify_static.c:66]: __INTEL_COMPILER is a macro which is only defined as compiler version when intel compiler is used, so seems `!defined`  is better? 
https://stackoverflow.com/questions/13534032/how-to-detect-intels-compiler-icc-with-ifdef

Comment (usstq) [reply src/common/ittnotify/ittnotify_static.c:66]: I tested on VC2019 and GCC, as long as __INTEL_COMPILER is not defined as 0 or nothing, `!__INTEL_COMPILER` is the same as `!defined(__INTEL_COMPILER)` in practice, so I guess we can keep that

Comment (dzarukin) [reply src/common/ittnotify/ittnotify_static.c:66]: This is oneDNN custom change (haven't been promoted to ITT) which comes as warning if `-Wundef` is used for compilation.

Comment (usstq) [reply src/common/ittnotify/ittnotify_static.c:66]: Oh, I see, I also double confirmed it by checking all references to `__INTEL_COMPILER` in oneDNN master, all of them are `defined(__INTEL_COMPILER)` except this one, I guess it was actually changed by OpenVINO guys from `!__INTEL_COMPILER` to `defined(__INTEL_COMPILER)`  in our fork for a reason? 
@EgorDuplensky

**Comment by EgorDuplensky** [comment]
Fixed in scope of:
- https://github.com/openvinotoolkit/oneDNN/pull/166
- https://github.com/openvinotoolkit/openvino/pull/14252

**Pull request closed without merging**